src/mac-now-playing/MacNowPlaying.mm

```diff
@@ -0,0 +1,267 @@
+        ::String title = tuple.get_str(Tuple::Title);
+        String artist = tuple.get_str(Tuple::Artist);
+        String album = tuple.get_str(Tuple::Album);
+        String file = aud_drct_get_filename();
+        int current_time = aud_drct_get_time();
+        int length = tuple.get_int(Tuple::Length);
+
+        NSMutableDictionary<NSString *, id> * nowPlayingInfo =
+            [[NSMutableDictionary alloc] init];
+
+        nowPlayingInfo[MPNowPlayingInfoPropertyMediaType] =
+            [[NSNumber alloc] initWithInt:MPNowPlayingInfoMediaTypeAudio];
+        nowPlayingInfo[MPNowPlayingInfoPropertyPlaybackRate] =
+            [[NSNumber alloc] initWithBool:aud_drct_get_playing()];
+        nowPlayingInfo[MPMediaItemPropertyTitle] =
+            [[NSString alloc] initWithUTF8String:title ? title : ""];
+        nowPlayingInfo[MPMediaItemPropertyArtist] =
+            [[NSString alloc] initWithUTF8String:artist ? artist : ""];
+        nowPlayingInfo[MPMediaItemPropertyAlbumTitle] =
+            [[NSString alloc] initWithUTF8String:album ? album : ""];
+        nowPlayingInfo[MPMediaItemPropertyMediaType] =
+            [[NSNumber alloc] initWithInt:MPMediaTypeMusic];
+        nowPlayingInfo[MPMediaItemPropertyPlaybackDuration] =
+            [[NSNumber alloc] initWithInt:length / 1000];
+        nowPlayingInfo[MPNowPlayingInfoPropertyElapsedPlaybackTime] =
+            [[NSNumber alloc] initWithInt:current_time];
+
+        if (file)
+        {
+            AudArtPtr artwork = aud_art_request(file, AUD_ART_FILE);
+            const Index<char> * artworkBytes = artwork.data();
+            if (artworkBytes)
+            {
+                NSData * nsBytes =
+                    [[NSData alloc] initWithBytes:artworkBytes->begin()
+                                           length:artworkBytes->len()];
+                NSImage * image = [[NSImage alloc] initWithData:nsBytes];
+                MPMediaItemArtwork * mpArtwork = [[MPMediaItemArtwork alloc]
+                    initWithBoundsSize:[image size]
+                        requestHandler:^NSImage * _Nonnull(CGSize size) {
+                          return image;
+                        }];
+                nowPlayingInfo[MPMediaItemPropertyArtwork] = mpArtwork;
+            }
+        }
+        [center setNowPlayingInfo:nowPlayingInfo];
+    }
+}
+
+static void update_playback_status(void * data, void * user)
+{
+    remote_command_set_playback_state();
+    update_player_time();
+}
+
+const char MacNowPlayingPlugin::about[] =
+    N_("Now Playing Plugin for Audacious\n"
+       "Copyright 2025 Yves Ndiaye\n\n"
+       "This plugin implements the Apple Now Playing interface."
+       "It allows Audacious to be controlled throught media keys,"
+       "to show the current album artwork and to seek the playback.");
+
+bool MacNowPlayingPlugin::init()
+{
+    (void)remote_command_setup_handler();
+    (void)remote_command_start();
+    (void)remote_command_disable_commands();
```

Comment:
What was your intention here for the `(void)` prefixes? Remove them, they are useless when calling void functions.